src/main/java/org/javarosa/core/model/TriggerableDag.java

```diff
@@ -381,8 +379,6 @@ public void finalizeTriggerables(FormInstance mainInstance, EvaluationContext ev
                 throwCyclesInDagException(vertices);
 
             List<QuickTriggerable> orderedRoots = new ArrayList<>(roots);
-            // TODO Study if insertion order & using LinkedSets would suffice to ensure proper triggerable chaining
-            Collections.sort(orderedRoots, QuickTriggerableComparator.INSTANCE);
```

Comment:
I don't think you need `orderedRoots` at all, then, and can just `addAll(roots)`.